tests/validation/tests/v3_api/test_cert_rotation.py

```diff
@@ -0,0 +1,124 @@
+import pytest
+import datetime
+from .common import *  # NOQA
+from rancher import ApiError
+
+#Gets the certificate expiration date and cert name. Stores them in a dict.
+def get_certs():
+    certs={}
+    client, cluster = get_user_client_and_cluster()
+    for key in cluster.certificatesExpiration:
+        certs[key]=parse_datetime(cluster.certificatesExpiration[key]["expirationDate"])
+    #Delete etcd node certs
+    remove=[]
+    for key in certs:
+        if "kube-etcd" in key:
+            remove.append(key)
+    for i in remove:
+        certs.pop(i)
+    #Get etcd node certs from node IP
+    nodes=get_etcd_nodes(cluster)
+    for node in nodes:
+        if node["labels"]["node-role.kubernetes.io/etcd"] == "true":
+            ipKey="kube-etcd-"+node["ipAddress"].replace(".","-")
+            certs[ipKey]=parse_datetime(cluster.certificatesExpiration[ipKey]["expirationDate"])
+    return certs
+
+#Turn expiration string into datetime
+def parse_datetime(expiration_string):
+    return datetime.datetime(int(expiration_string[:4]),int(expiration_string[5:7]),int(expiration_string[8:10]),int(expiration_string[11:13]),int(expiration_string[14:16]),int(expiration_string[17:19]))
```

Comment:
Can you check if this can be converted to a time object and diff of the start and end time could be compared?